core/src/main/java/com/linecorp/armeria/internal/SslContextUtil.java

```diff
@@ -65,20 +71,45 @@
 
     public static final List<String> DEFAULT_PROTOCOLS = ImmutableList.of("TLSv1.3", "TLSv1.2");
 
+    private static final List<String> DEFAULT_JDKENGINE_CIPHERS = ImmutableList.<String>builder()
+            .addAll(Http2SecurityUtil.CIPHERS)
+            .build();
+
+    private static final List<String> DEFAULT_JDKENGINE_PROTOCOLS = ImmutableList.of("TLSv1.2");
+
     /**
-     * Creates a {@link SslContext} with Armeria's defaults, enabling support for HTTP/2, TLSv1.3, and
-     * TLSv1.2.
+     * Creates a {@link SslContext} with Armeria's defaults, enabling support for HTTP/2,
+     * TLSv1.3 (if supported), and TLSv1.2.
      */
     public static SslContext createSslContext(Supplier<SslContextBuilder> sslContextSupplier,
                                               boolean forceHttp1,
                                               Consumer<? super SslContextBuilder> userCustomizer) {
         final SslContextBuilder builder = sslContextSupplier.get();
 
-        builder.sslProvider(Flags.useOpenSsl() ? SslProvider.OPENSSL : SslProvider.JDK)
-                  .protocols(DEFAULT_PROTOCOLS.toArray(new String[0]));
-
-        builder.ciphers(DEFAULT_CIPHERS, SupportedCipherSuiteFilter.INSTANCE);
-
+        if (Flags.useOpenSsl()) {
```

Comment:
We don't need to branch on this flag since the below check will still work correctly for openssl and is more precise. Basically, we want to do feature detection instead of version detection.